internal/ssa2ast/polyfill.go

```diff
@@ -0,0 +1,288 @@
+package ssa2ast
+
+import (
+	"go/ast"
+	"go/token"
+	"go/types"
+)
+
+func makeMapIteratorPolyfill(tc *typeConverter, mapType *types.Map) (ast.Expr, types.Type, error) {
+	keyTypeExpr, err := tc.Convert(mapType.Key())
+	if err != nil {
+		return nil, nil, err
+	}
+	valueTypeExpr, err := tc.Convert(mapType.Elem())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	nextType := types.NewSignatureType(nil, nil, nil, nil, types.NewTuple(
+		types.NewVar(token.NoPos, nil, "", types.Typ[types.Bool]),
+		types.NewVar(token.NoPos, nil, "", mapType.Key()),
+		types.NewVar(token.NoPos, nil, "", mapType.Elem()),
+	), false)
+
+	// Generated using https://github.com/lu4p/astextract from snippet:
+	/*
+		func(m map[<key type>]<value type>) func() (bool, <key type>, <value type>) {
+			keys := make([]<key type>, 0, len(m))
+			for k := range m {
+				keys = append(keys, k)
+			}
+			i := 0
+			return func() (ok bool, k <key type>, r <value type>) {
+				if i < len(keys) {
+					k = keys[i]
+					ok, r = true, m[k]
+					i++
+				}
+				return
+			}
+		}
+	*/
```

Comment:
understood that this is generated, but these 200 lines could really be 100 or even 50 lines. Some lines just have two non-whitespace characters. maybe we should teach astextract do do a better job?

Reply:
it's one case where a polyfill is needed (and hopefully in the future there will be a public iterator for maps), so i just optimize code manually

Reply:
I think the generated code is ugly, but its just generated code I don't think it really matters. 
I had similar problems with ugly generated code before, @mvdan maybe gofumpt can put struct instantiations with a single field on one line? 

Regex to save 50 lines https://regexr.com/7fkvj

Reply:
I don't think gofumpt could/should do this, because it's designed to not have any options, and by default it shouldn't be this aggressive in joining lines. Maybe someone does want the extra newlines, if they intend to add more fields later on, or if they're still editing the code. For the same reason, gofumpt doesn't turn

```
import (
    "foo"
)
```

into `import "foo"`.